### PR TITLE
Test the full matrix for polars and dask wheels on nightlies

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This PR ensures that we have nightly coverage of more of the CUDA/Python/arch versions that we claim to support for dask-cudf and cudf-polars wheels.

In addition, this PR ensures that we do not attempt to run the dbgen executable in the Polars repository on systems with too old of a glibc to support running them.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
To validate that the new matrix entries will run successfully I modified pr.yaml to run the tests in PR CI once as well. See the CI run in https://github.com/rapidsai/cudf/pull/17320/commits/cb215484c2b1538a47c4b835d47d9a2d32bac4a3 for that verification.

For more information on the glibc change, see https://github.com/rapidsai/cudf/pull/17336.

File: ci/run_cudf_polars_polars_tests.sh [hunk 1]
```diff
@@ -14,6 +14,24 @@ DESELECTED_TESTS=(
     "tests/docs/test_user_guide.py" # No dot binary in CI image
 )
 
+if [[ $(arch) == "aarch64" ]]; then
+    # The binary used for TPC-H generation is compiled for x86_64, not aarch64.
+    DESELECTED_TESTS+=("tests/benchmark/test_pdsh.py::test_pdsh")
+    # The connectorx package is not available on arm
+    DESELECTED_TESTS+=("tests/unit/io/database/test_read.py::test_read_database")
+    # The necessary timezone information cannot be found in our CI image.
```
Comment (bdice) [inline ci/run_cudf_polars_polars_tests.sh:22]: What timezone info are we missing?

**Review by bdice — APPROVED** [review]
One question but all looks fine to me.

File: ci/run_cudf_polars_polars_tests.sh (hunk 1 above)
Comment (vyasr) [reply ci/run_cudf_polars_polars_tests.sh:22]: You can see the failures in https://github.com/rapidsai/cudf/actions/runs/11846396105/job/33015080248

**Pull request merged**